src/compiler/checker.ts

```diff
@@ -31899,7 +31900,7 @@ namespace ts {
 
             // Abstract methods cannot have an implementation.
             // Extra checks are to avoid reporting multiple errors relating to the "abstractness" of the node.
-            if (hasSyntacticModifier(node, ModifierFlags.Abstract) && node.kind === SyntaxKind.MethodDeclaration && node.body) {
+            if (hasSyntacticModifier(node, ModifierFlags.Abstract) && node.kind === SyntaxKind.MethodDeclaration && node.body && !isInJSFile(node)) {

```

Comment:
I wondered why this change is necessary as there cannot be syntactic modifiers in JS files. In case you also wondered: it turns out this might be a bug: #42189

Reply:
`hasSyntacticModifier` does not include `JSDoc`.